apps/happy-blocks/src/support-page/edit.tsx

```diff
@@ -0,0 +1,101 @@
+import './edit.scss';
+import { BlockControls, useBlockProps } from '@wordpress/block-editor';
+import { BlockEditProps } from '@wordpress/blocks';
+import { ToolbarButton, ToolbarGroup, withNotices } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { edit } from '@wordpress/icons';
+import React from 'react';
+import { fetchPageAttributes, SUPPORT_PAGE_PATTERN, SupportPageBlockAttributes } from './block';
+import { EmbedPlaceHolder } from './embed-placeholder';
+import { WordPressIcon } from './icon';
+import { SupportPageEmbed } from './support-page-embed';
+
+type EditProps = BlockEditProps< SupportPageBlockAttributes > &
+	withNotices.Props & {
+		noticeUI: JSX.Element;
+	};
+
+/**
+ * The edit function describes the structure of your block in the context of the
+ * editor. This represents what the editor will render when the block is used.
+ *
+ * @see https://developer.wordpress.org/block-editor/reference-guides/block-api/block-edit-save/#edit
+ * @returns {import("@wordpress/element").WPElement} Element to render.
+ */
+export const Edit = compose( withNotices )( ( props: EditProps ) => {
+	const { attributes, className, setAttributes, noticeOperations, noticeUI } = props;
+
+	const instructions = __( 'Paste a link to the page you want to display.', 'happy-blocks' );
+	const mismatchErrorMessage = __( 'It does not look like an embeddable URL.', 'happy-blocks' );
+	const placeholder = __( 'Enter URL to embed here…', 'happy-blocks' );
+	const label = __( 'WordPress Guide page URL', 'happy-blocks' );
+
+	const [ editing, setEditing ] = useState( false );
+	const [ url, setUrl ] = useState( attributes.url );
+
+	const onEditModeToggle = () => {
+		setEditing( ! editing );
+	};
+
+	const onSubmit = async () => {
+		if ( ! SUPPORT_PAGE_PATTERN.test( url ) ) {
+			noticeOperations.removeAllNotices();
+			noticeOperations.createErrorNotice( mismatchErrorMessage );
+			return;
+		}
+
+		try {
+			setAttributes( { url } );
+
+			const fetchedAttributes = await fetchPageAttributes( url );
+
+			noticeOperations.removeAllNotices();
+			setEditing( false );
+
+			setAttributes( fetchedAttributes );
+		} catch ( e: any ) {
+			noticeOperations.removeAllNotices();
+			noticeOperations.createErrorNotice(
+				e.message || e || __( 'Unable to fetch the page, check the URL', 'happy-blocks' )
+			);
+		}
+	};
+
+	const blockProps = useBlockProps( {
+		className: 'wp-block-p2-embed',
```

Comment:
I'm wondering whether we should prefix this with `wp-block-support-page-embed` or even `wp-block-happy-blocks-embed` instead. What do you think?

Reply:
Right, it has slipped out of P2 codebase which I used as the base :)

Reply:
> I'm wondering whether we should prefix this with wp-block-support-page-embed or even wp-block-happy-blocks-embed instead

I used `hb-support-page-embed-placeholder`, where `hb` stands for `happyblocks` app inside Callypso.

Reply:
Alright 👍